### PR TITLE
SmallRye health checks - Singleton as the default scope

### PR DESCRIPTION
**Pull request by mkouba**
- small refactoring - reuse code and get rid of some boilerplate
- update docs

### CODE REVIEW

File: docs/src/main/asciidoc/microprofile-health.adoc [hunk 1]
```diff
@@ -124,10 +124,10 @@ public class SimpleHealthCheck implements HealthCheck {
     }
 }
 ----
+<1> It's recommended to annotate the health check class with `@ApplicationScoped` or the `@Singleton` scope so that a single bean instance is used for all health check requests. 
+<2> If a bean class annotated with one of the health check annotations declares no scope then the `@Singleton` scope is used automatically.
 
-As you can see health check procedures are defined as implementations of the 
-`HealthCheck` interface which are defined as CDI beans with the one of the 
-following CDI qualifiers: 
+As you can see, the health check procedures are defined as CDI beans that implementat the `HealthCheck` interface and are annotated with the one of the health check qualifiers, such as:
```
Comment (kenfinnigan) [inline docs/src/main/asciidoc/microprofile-health.adoc:130]: "implementat" should be "implement"?

Comment (kenfinnigan) [reply docs/src/main/asciidoc/microprofile-health.adoc:130]: Maybe change "with the one of" to "with one of"

**Review by kenfinnigan — APPROVED** [review]
Liked the abstraction for the handlers, nice

File: docs/src/main/asciidoc/microprofile-health.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/microprofile-health.adoc:130]: Yep, sure. Thanks for spotting these!

**Review by cescoffier — APPROVED** [review]

**Comment by mkouba** [comment]
```
2020-06-04T20:12:46.4263438Z [ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.128 s <<< FAILURE! - in io.quarkus.vertx.http.security.FormAuthCookiesTestCase
2020-06-04T20:12:46.4264511Z [ERROR] testCredentialCookieRotation  Time elapsed: 1.267 s  <<< FAILURE!
2020-06-04T20:12:46.4268688Z org.opentest4j.AssertionFailedError: Having to wait for -1 ms for another request is unexpected. The previous one took too long. ==> expected: <true> but was: <false>
```
@Karm I wonder if we should add some more logging or even relax the time limits a bit?

**Pull request merged**